### PR TITLE
Preserve options when doing a cache push

### PR DESCRIPTION
**Pull request by carlossg**
Otherwise options like `insecure` are lost

### CODE REVIEW

**Comment by carlossg** [comment]
For the record this moves forward allowing insecure locally deployed docker registries but doesn't make it work yet, the builds still can't find the cached image even though it is in the repo

File: pkg/executor/push.go [hunk 1]
```diff
@@ -126,7 +126,7 @@ func pushLayerToCache(opts *config.KanikoOptions, cacheKey string, layer v1.Laye
 	if err != nil {
 		return errors.Wrap(err, "appending layer onto empty image")
 	}
-	return DoPush(empty, &config.KanikoOptions{
-		Destinations: []string{cache},
-	})
+	cacheOpts := opts
+	cacheOpts.Destinations = []string{cache}
```
Comment (dlorenc) [inline pkg/executor/push.go]: This mutates opts, could we do it by creating a new object?

Comment (carlossg) [reply pkg/executor/push.go]: sorry, I think I've fixed it

Comment (priyawadhwa) [reply pkg/executor/push.go]: I think this should work from this [test](https://play.golang.org/p/Nld2igwDRAx)

**Pull request merged**